components/org.wso2.carbon.identity.api.server.secret.management/org.wso2.carbon.identity.api.server.secret.management.v1/src/main/java/org/wso2/carbon/identity/api/server/secret/management/v1/core/SecretManagementService.java

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.identity.api.server.secret.management.v1.core;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.api.server.common.error.APIError;
+import org.wso2.carbon.identity.api.server.common.error.ErrorResponse;
+import org.wso2.carbon.identity.api.server.secret.management.common.SecretManagementConstants;
+import org.wso2.carbon.identity.api.server.secret.management.common.SecretManagementServiceHolder;
+import org.wso2.carbon.identity.api.server.secret.management.v1.model.Secret;
+import org.wso2.carbon.identity.api.server.secret.management.v1.model.SecretAdd;
+import org.wso2.carbon.identity.api.server.secret.management.v1.model.SecretUpdateRequest;
+import org.wso2.carbon.identity.secret.mgt.core.exception.SecretManagementClientException;
+import org.wso2.carbon.identity.secret.mgt.core.exception.SecretManagementException;
+import org.wso2.carbon.identity.secret.mgt.core.exception.SecretManagementServerException;
+import org.wso2.carbon.identity.secret.mgt.core.model.Secrets;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.ws.rs.core.Response;
+
+import static org.wso2.carbon.identity.api.server.secret.management.common.SecretManagementConstants.CONFIG_MGT_ERROR_CODE_DELIMITER;
+import static org.wso2.carbon.identity.api.server.secret.management.common.SecretManagementConstants.ERROR_PREFIX;
+
+/**
+ * Invoke internal OSGi service to perform secret management operations.
+ */
+public class SecretManagementService {
+
+    private static final Log log = LogFactory.getLog(SecretManagementService.class);
+
+    /**
+     * Create a secret.
+     *
+     * @param secretAdd secret post request.
+     * @return secret.
+     */
+    public Secret addSecret(String secretType, SecretAdd secretAdd) {
+
+        validateSecretAdd(secretAdd);
+        org.wso2.carbon.identity.secret.mgt.core.model.Secret requestDTO, responseDTO;
```

Comment:
why we need to have full qualified package name

Reply:
Resolved this.
previously class name Secret is used for both repos. Now change the Secret -> SecretResponse in this repo